### PR TITLE
return bad request instead of server error for identity group cycle detection

### PR DESCRIPTION
**Pull request by ccapurso**
Providing `member_group_ids` that contains a cycle for an identity group to `identity/group/name/<name>` will currently result in a 500 response. This should instead be a 400 response stating that a cycle was detected.

**Reproduction steps:**

Create groups:

```
ID_Group_0=`vault write -format=json identity/group/name/Group_0 | jq -r ".data.id"
ID_Group_1=`vault write -format=json identity/group/name/Group_1 | jq -r ".data.id"
ID_Group_2=`vault write -format=json identity/group/name/Group_2  | jq -r ".data.id"
```

Specify group IDs for `Group_0` and `Group_1`:
```
vault write identity/group/name/Group_0 member_group_ids= 
vault write identity/group/name/Group_1 member_group_ids=$ID_Group_2 
```

Specify group IDs for `Group_2` will fail due to cycle:
```
vault write identity/group/name/Group_2 member_group_ids=$ID_Group_0,$ID_Group_1 

Error writing data to identity/group/name/Group_2: Error making API request.

URL: PUT http://127.0.0.1:8200/v1/identity/group/name/Group_2
Code: 500. Errors:

1 error occurred:
	* failed to perform cyclic relationship detection for member group ID "c5ac421e-0748-8b7e-466b-5a951542c2f8"
```

This PR introduces improved error handling such that an error caused by a cycle detection can be handled more specifically by a request handler. This allows for returning a `logical.ErrorResponse` rather than bubbling up the error to Vault's response handling logic and resulting in a 500 as so:

```
vault write identity/group/name/Group_2 member_group_ids=$ID_Group_0,$ID_Group_1 

Error writing data to identity/group/name/Group_2: Error making API request.

URL: PUT http://127.0.0.1:8200/v1/identity/group/name/Group_2
Code: 400. Errors:

* cyclic relationship detected for member group ID "752b6146-2cc3-b3f5-8ab6-9feea4da98b5"
```

### CODE REVIEW

File: changelog/15912.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:change
+identity: a request to `/identity/group` that includes `member_group_ids` that contains a cycle will now be responded to with a 400 rather than 500
```
Comment (hghaf099) [inline changelog/15912.txt:2]: Should we be concerned with backwards compatibility?

**Comment by ccapurso** [comment]
It looks like some tests failed, looking into those.

**Review by hghaf099 — APPROVED** [review]
Asked a question, otherwise looks good!

File: changelog/15912.txt (hunk 1 above)
Comment (ccapurso) [reply changelog/15912.txt:2]: That is mainly why I listed this as a `change`. Our [API docs](https://www.vaultproject.io/api-docs#http-status-codes) mention that a 500 response can be retried. Retrying this request will _always_ fail in its current state. Either the provided `member_group_ids` need to change or a separate request against one of the contained groups will need to be made to remove the potential cycle.

Comment (hghaf099) [reply changelog/15912.txt:2]: Thanks for the explanation. Make sense!

**Pull request merged**